wordpress-mobile/WordPress-iOS
# Display a more appropriate message for invalid queries

Fixes #15155

#### Related PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/489

### Description

This PR improves the error message for invalid domain queries (e.g. when invalid characters are in the query string). The approach is very similar to the [equivalent Android PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15969)

### To Test
1. Start creating a new site using the app (From My Site screen tap the ▼ icon next to the site title)
2. On the Site Picker screen tap on the ➕ menu button
3. Select **Create WordPress.com site**
4. Choose any design or tap the **SKIP** menu button
5. Type `_` or any word with non-English characters
6. **Verify** that the text message under the input is:
   > Your search includes characters not supported in WordPress.com domains. The following characters are allowed: A–Z, a–z, 0–9.


## Regression Notes
1. Potential unintended areas of impact
Nothing comes to mind

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Automated tests for this are included in the related WordPressKit-iOS PR.

PR submission checklist:
